lib/src/events.dart

```diff
@@ -1,120 +1,246 @@
 import 'package:flutter_webrtc/flutter_webrtc.dart' as rtc;
+import 'package:livekit_client/livekit_client.dart';
 
+import 'participant/participant.dart';
+import 'participant/remote_participant.dart';
 import 'proto/livekit_models.pb.dart' as lk_models;
+import 'proto/livekit_rtc.pb.dart' as lk_rtc;
+import 'track/remote_track_publication.dart';
+import 'track/track.dart';
+import 'types.dart';
 
-abstract class LiveKitEvent {}
-
-abstract class RoomEvent implements LiveKitEvent {
-  const RoomEvent();
-}
-
-abstract class ParticipantEvent implements LiveKitEvent {
-  const ParticipantEvent();
-}
-
-abstract class EngineEvent implements LiveKitEvent {
-  const EngineEvent();
-}
-
-abstract class TrackEvent implements LiveKitEvent {
-  const TrackEvent();
+abstract class LiveKitEvent {
+  const LiveKitEvent();
 }
 
 //
 // Room events
 //
-class RoomReconnectingEvent extends RoomEvent {}
 
-class RoomReconnectedEvent extends RoomEvent {}
+/// When the connection to the server has been interrupted and it's attempting
+/// to reconnect.
+class RoomReconnectingEvent extends LiveKitEvent {}
 
-class RoomDisconnectedEvent extends RoomEvent {}
+/// Connection to room is re-established. All existing state is preserved.
+class RoomReconnectedEvent extends LiveKitEvent {}
 
-class RoomParticipantConnectedEvent extends RoomEvent {}
+/// Disconnected from the room
+class RoomDisconnectedEvent extends LiveKitEvent {}
 
-class RoomParticipantDisconnectedEvent extends RoomEvent {}
+/// When a new [RemoteParticipant] joins *after* the current participant has connected
+/// It will not fire for participants that are already in the room
+class RoomParticipantConnectedEvent extends LiveKitEvent {
+  final RemoteParticipant participant;
+  const RoomParticipantConnectedEvent({
+    required this.participant,
+  });
+}
 
-class RoomTrackPublishedEvent extends RoomEvent {}
+/// When a [RemoteParticipant] leaves the room
+class RoomParticipantDisconnectedEvent extends LiveKitEvent {
+  final RemoteParticipant participant;
+  const RoomParticipantDisconnectedEvent({
+    required this.participant,
+  });
+}
 
-class RoomTrackSubscribedEvent extends RoomEvent {}
+/// When a new track is published to room *after* the current participant has
+/// joined. It will not fire for tracks that are already published
+// (Relayed from LiveKitEvent)
+class RoomTrackPublishedEvent extends LiveKitEvent {}
+
+/// A [RemoteParticipant] has unpublished a track
+// (Relayed from LiveKitEvent)
+class RoomTrackUnpublishedEvent extends LiveKitEvent {}
+
+/// A subscribed track is no longer available.
```

Comment:
Events look really clean! My only comment is if there's a way to make it more clear which events are fired on Room, and which on Participant. Can we add a line indicating that? i.e. `emitted by Room and Participant` ?

Reply:
Yea that's confusing! will clean it up!